### DISCUSSION TITLE
issue with open-webui not listing the models of ollama[solved]

### DISCUSSION THREAD

**Discussion by ds47x** [open-webui/open-webui#4376] [General]
# Bug Report
The issue is when trying to select a model the drop down menu says no results found 

## Description
The issue is i cant select or find  llama models on the webui

>i checked ollama if it is running
```
curl http://127.0.0.1:11434/

Ollama is running⏎ 
```  

```
ollama list
NAME                            ID              SIZE    MODIFIED     
llama3.1:latest                 62757860e01    4.7 GB  2 hours ago 
llama2-uncensored:latest        44040b92233    3.8 GB  2 months ago
```
ollama works on the terminal but doesnt get detected on the webui

**Steps to Reproduce:**
1.install ollama
`curl -fsSL https://ollama.com/install.sh | sh`

install any model
`ollama run llama3.1`

then download webui using docker

`sudo docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main`

and done but the models are not detected and it said server error when trying to download models by the webui

```
INFO  [apps.openai.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]

```

**Expected Behavior:**
what i expected to happen was download the webui and use the llama models on it

**Actual Behavior:**
the models are not listed on the webui

## Environment

- **Open WebUI Version:**v0.3.11
- **Ollama (if applicable):** 0.1.39
- **Operating System:** EndeavorsOS  
- **Browser (if applicable):firefox 128.0.3

**Confirmation:**

- [ y] I have read and followed all the instructions provided in the README.md.
- [ y] I am on the latest version of both Open WebUI and Ollama.
- [ y] I have included the browser console logs.
- [ y] I have included the Docker container logs.

## Logs and Screenshots
![ol](https://github.com/user-attachments/assets/fa9a457a-167f-40f6-85a9-92bc5e9f6c7c)
![oll](https://github.com/user-attachments/assets/9e372ca3-5c34-4991-8c41-2ea83ba155c4)

**Docker Container Logs:**
```
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
INFO  [apps.openai.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
INFO  [apps.openai.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
ERROR:    Exception in ASGI application
  + Exception Group Traceback (most recent call last):
  |   File "/usr/local/lib/python3.11/site-packages/starlette/_utils.py", line 87, in collapse_excgroups
  |     yield
  |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 190, in __call__
  |     async with anyio.create_task_group() as task_group:
  |   File "/usr/local/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 680, in __aexit__
  |     raise BaseExceptionGroup(
  | ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 435, in run_asgi
    |     result = await app(  # type: ignore[func-returns-value]
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/usr/local/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    |     return await self.app(scope, receive, send)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__
    |     await super().__call__(scope, receive, send)
    |   File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 123, in __call__
    |     await self.middleware_stack(scope, receive, send)
    |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 186, in __call__
    |     raise exc
    |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 164, in __call__
    |     await self.app(scope, receive, _send)
    |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 189, in __call__
    |     with collapse_excgroups():
    |   File "/usr/local/lib/python3.11/contextlib.py", line 158, in __exit__
    |     self.gen.throw(typ, value, traceback)
    |   File "/usr/local/lib/python3.11/site-packages/starlette/_utils.py", line 93, in collapse_excgroups
    |     raise exc
    |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 191, in __call__
    |     response = await self.dispatch_func(request, call_next)
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/app/backend/main.py", line 902, in update_embedding_function
    |     response = await call_next(request)
    |                ^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 166, in call_next
    |     raise RuntimeError("No response returned.")
    | RuntimeError: No response returned.
    +------------------------------------

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 435, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 123, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 186, in __call__
    raise exc
  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 189, in __call__
    with collapse_excgroups():
  File "/usr/local/lib/python3.11/contextlib.py", line 158, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/usr/local/lib/python3.11/site-packages/starlette/_utils.py", line 93, in collapse_excgroups
    raise exc
  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 191, in __call__
    response = await self.dispatch_func(request, call_next)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backend/main.py", line 902, in update_embedding_function
    response = await call_next(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/base.py", line 166, in call_next
    raise RuntimeError("No response returned.")
RuntimeError: No response returned.
INFO  [apps.openai.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
INFO  [apps.openai.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
INFO:     172.17.0.1:52882 - "GET /_app/immutable/nodes/7.b46cc32c.js HTTP/1.1" 200 OK
INFO  [apps.openai.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
INFO:     172.17.0.1:52882 - "GET /api/v1/users/ HTTP/1.1" 200 OK
INFO  [apps.openai.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
INFO  [apps.openai.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
INFO:     172.17.0.1:52894 - "GET /api/webhook HTTP/1.1" 200 OK
INFO:     172.17.0.1:52882 - "GET /api/v1/auths/admin/config HTTP/1.1" 200 OK
INFO  [apps.openai.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
INFO  [apps.ollama.main] get_all_models()
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
INFO:     172.17.0.1:52882 - "GET /ollama/config HTTP/1.1" 200 OK
INFO  [apps.openai.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
INFO  [apps.openai.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
INFO  [apps.ollama.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
INFO:     172.17.0.1:52894 - "GET /ollama/api/version HTTP/1.1" 500 Internal Server Error
INFO:     172.17.0.1:52882 - "GET /ollama/urls HTTP/1.1" 200 OK
INFO  [apps.openai.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
INFO  [apps.ollama.main] get_all_models()
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
INFO  [apps.ollama.main] url: http://host.docker.internal:11434
INFO:     172.17.0.1:52882 - "POST /ollama/api/pull/0 HTTP/1.1" 500 Internal Server Error
INFO  [apps.openai.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
INFO  [apps.openai.main] get_all_models()
INFO  [apps.ollama.main] get_all_models()
ERROR [apps.ollama.main] Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]


```

## Installation Method

Docker 

!

**Comment by ds47x** [comment]
i tried updating ollama to 0.3.3 but still the same. tried using 
`docker run -d --network=host -v open-webui:/app/backend/data -e OLLAMA_BASE_URL=http://127.0.0.1:11434 --name open-webui --restart always ghcr.io/open-webui/open-webui:main`
but still the same.

again using` Environment="ollama_host=0.0.0.0"` but nothing changed

**Reply by pavelfialkovaky** [reply]
-network=host thank you

**Comment by ds47x** [comment]
i was dumb enough not to see this, sorry for the whole trouble i found the solution in the readme.md

Open WebUI: Server Connection Error
If you're experiencing connection issues, it’s often due to the WebUI docker container not being able to reach the Ollama server at 127.0.0.1:11434 (host.docker.internal:11434) inside the container . Use the --network=host flag in your docker command to resolve this. Note that the port changes from 3000 to 8080, resulting in the link: `http://localhost:8080`.


`docker run -d --network=host -v open-webui:/app/backend/data -e OLLAMA_BASE_URL=http://127.0.0.1:11434 --name open-webui --restart always ghcr.io/open-webui/open-webui:main`


the port will be changed to 8080 so `http://localhost:8080`.

**Reply by CHAOJICHENG5** [reply]
thanks a lot

**Reply by architectonio** [reply]
Same problem for me...since I upgraded Ollama and Open-Webui.

Unfortunately the solution is not applicable for me cause the the port 8080 on my Server is already busy (used by another service)
Is there any way to change the port 8080 of Open-Webui to another one, lets say 48080?
Thanks

**Reply by haebler** [reply]
Pass a different port to the container e.g. `-e PORT=48080`
Reference: https://docs.openwebui.com/getting-started/env-configuration#port

**Reply by architectonio** [reply]
> Pass a different port to the container e.g. `-e PORT=48080` Reference: https://docs.openwebui.com/getting-started/env-configuration#port

Thanks a lot!

**Reply by wxsabi** [reply]
Absolute LEGEND!

**Reply by tejasvih** [reply]
Thanks @ds47x and @architectonio it works with independent installation of ollama and dockerized open-webui.
The command I am using for this is:
`docker run -d -e PORT=48080 --gpus all --network=host -e OLLAMA_BASE_URL=http://127.0.0.1:11434 -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:cuda`

**Reply by jjfumero** [reply]
Thanks for posting the solution. I had the same problem

**Reply by liuzhch1** [reply]
For anyone still failed to access ollama even with `--network=host` in macOS. Remember to enable host networking in Docker Desktop: Settings/Resources/Network/Enable host networking.
https://stackoverflow.com/a/78531505

**Reply by damilarelana** [reply]
works for me also ... (after many different iterations and permutations)

**Reply by fpscit** [reply]
you are a legend . i was stuck on this issue for last 2 days . finally its working . i can see my ollama models. before i was using it at 3000 port . now its 8080 , how can i go back at 3000 any idea .

**Reply by mikwee** [reply]
Does this apply to Podman too? I'm having the same issue there, but the docs say no special flag is needed for `pasta`.

**Reply by FirthR5** [reply]
Thank you.
I do a variation because a really f***ing hate 8080 port and for people who don't like launching on the 8080:
```
docker rm -f open-webui
docker run -d --network=host -v open-webui:/app/backend/data -e OLLAMA_BASE_URL=http://127.0.0.1:11434 -e PORT=10001 --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```

**Comment by jzhang82119** [comment]
Thanks a lot.

**Comment by rotleaf** [comment]
I am still having this problem, tried the solutions here, nothing works,

**Reply by haebler** [reply]
@rotleaf Would you be able to provide some data on how you are trying to set this up? Without that it is really hard to help you.

**Comment by jasonculligan** [comment]
+1 for thanks - the documentation needs to be corrected.

**Comment by rymapu** [comment]
Dave's Garage uses:
docker run -d -p 3000:8080 --gpus=all -v ollama:/root/.ollama -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:ollama

I'm here because it's not working, process is mostly the same as above, I don't try to download. I have ran 3.1 and 3.2 from CLI
127.0.0.1:11434 never shows"... service running". always unreachable.

systemctl status ollama.service shows port 11434

**Reply by justinh-rahb** [reply]
That video assumes a clean install with no pre-existing Ollama CLI or downloaded models. It has a bundled copy of Ollama, and specifies a Docker volume to store its models in. Use a bind mount instead, and point it to the host directory your models are in:

```sh
docker run -d -p 3000:8080 --gpus=all -v /usr/share/ollama/.ollama:/root/.ollama -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:ollama
```

**Reply by pythagorean833** [reply]
This is in fact the issue.  I enjoy Dave's Garage and got here same as everybody because I just followed his instructions half ignorantly.  I already had installed ollama and run models on the command line and then tried dave's snap install docker and docker run suggestion, which failed for several reasons.  First off, don't do the snap install but rather just do the normal apt install.  If you are running this on ubuntu (not wsl like dave did) then you would also need to install some nvidia stuff to get the gpus=all flag to work and then, like @justinh-rahb says... you are doing a bind mount to load the models you already pulled into docker.  In my case because I was just using plain old normal ubuntu 24.04 when I installed all of the ollama stuff my models are in /home/myname/.ollama and so I am mounting that directory in root/.ollama in docker, i.e. -v /home/myname/.ollama:/root/.ollama

And lemme tell ya, I felt a genuine sense of glee when I logged into the open-webui and finally saw those darned models in there AND it actually used the gpu to answer a prompt.  So, thanks.

**Comment by Freeforyall** [comment]
I found that when I start open-webui and then try to see models, none appear. However, if I open a command prompt and then type ollama list, then go back to the browser and refresh, everything appears. I had originally thought that by changing the url to 127.0.0.1 or 192.168.x.x that it fixed my problem, but the ollama list command seems to be the solution for me. I hope this helps!

**Reply by RVoelk** [reply]
I have a slightly different issue.  Only some of the models on my remote Ollama server are visible in the dropdown.

**Reply by rcarmo** [reply]
I'm scratching my head about this as well, since I run Ollama remotely. I just can't get them to appear consistently in the UI.

**Comment by DevKaranMehta** [comment]
same issue , because openWebUi is working inside the container and ollama is not 

Solution works :


```
docker stop open-webui
docker rm open-webui

docker run -d -p 8080:8080 -v open-webui:/app/backend/data \
  -e OLLAMA_BASE_URL=http://host.docker.internal:11434 \
  --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```

 **Verify if OpenWebUI Can Reach Ollama**

`docker exec -it open-webui curl -I http://host.docker.internal:11434
`

**Reply by edisun3003** [reply]
i Tried this but still got error 
curl: (6) Could not resolve host: host.docker.internal:11434

**Reply by RashiqAzhan** [reply]
It doesn't even print anything. `--verbose` flag shows it as trying.
```bash
root@adbd423aa405:/app/backend#  curl --verbose http://host.docker.internal:11434
*   Trying 172.17.0.1:11434...
```

What gives?
Ollama is saying it's running and I don't see errors in that container.

**Comment by andrewlee1807** [comment]
This issue happended because of Ollama and Open-webUI cannot communicate.
**Solution**:
Step 1: Command to open Ollama service file: `sudo nano /etc/systemd/system/ollama.service`
Step 2: Add variable: `Environment="OLLAMA_HOST=172.17.0.1"` 
_Note: You can get this IP by looking for gateway IP address command: `sudo docker network inspect bridge`_
Step 3: Restart services:

```
sudo systemctl daemon-reload  #Reload the systemd configuration
sudo systemctl restart ollama  #Restart the Ollama service
sudo docker rm -f webui #Stop and remove OpenWebUI, you can check your container name is webiui or open-webui by using sudo docker ps -a
sudo docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```

You can check it from this [video](https://youtu.be/TsTJVd9LciY?si=X-xPIQ2LRtWrgYlu&t=241)

**Reply by edisun3003** [reply]
Still i got same error
curl: (7) Could not connect to host.docker.internal port 11434 after 0ms : couldn't connect to server

**Comment by ondiekisteven** [comment]
Here's what worked for me: 
if you have both ollama and open-webui in docker, then ensure they have a network. without the network I was getting the error, but after adding the network the error disappeared. 


```yml
networks:
  ollamanet:

services:
  ollama:
    image: ollama/ollama:latest
    container_name: ollama
    restart: unless-stopped
    ports:
      - "11434:11434"
    volumes:
      - /opt/ollama_data:/root/.ollama
    environment:
      - OLLAMA_HOST=0.0.0.0
    networks:
      - ollamanet

  open-webui:
    image: ghcr.io/open-webui/open-webui:latest
    container_name: open-webui
    environment:
      - OLLAMA_BASE_URL=http://ollama:11434
    ports:
      - "8080:8080"
    volumes:
      - /opt/open-webui:/app/backend/data
    restart: unless-stopped
    networks:
      - ollamanet

```

**Comment by RashiqAzhan** [comment]
I wanted to have separate docker compose file for the ollama and the web ui. Isn't this supposed to be working?

```yaml
services:
  open-webui:
    image: ghcr.io/open-webui/open-webui:main
    container_name: open-webui
    restart: unless-stopped
    environment:
      - OLLAMA_BASE_URL=http://host.docker.internal:11434
    ports:
      - 11000:8080
    extra_hosts:
      - host.docker.internal:host-gateway
    volumes:
      - open-webui:/app/backend/data

volumes:
  open-webui:
```

```yaml
services:
  ollama:
    image: ollama/ollama
    container_name: ollama
    restart: unless-stopped
    ports:
      - "11434:11434"
    volumes:
      - ./root/.ollama:/root/.ollama
```

Assuming I'm not misunderstanding what `host.docker.internal` is doing which should be exposing the services hosted on my local host machine to the container.

**Reply by RashiqAzhan** [reply]
```yaml
environment:
  - OLLAMA_BASE_URL=http://host.docker.internal:11434
```

Even this is not working.

A similar such line is working fine for another docker program just fine. Not sure why it's not working for open-webui.

**Reply by luckylinux** [reply]
My understanding is the opposite: `host.docker.internal` (Podman: `host.containers.internal`) is used, from inside the Container, in order to access Resources located on the HOST.

So it's effectively the Reverse as a normal Port Mapping.

- Port Mapping: from Host (or other LAN Clients, depending on IP Binding) ----> access Resources located within the Container
- `host.docker.internal` (Podman: `host.containers.internal`): from within Container ----> access Resources located on the Host

`0.0.0.0` or `127.0.0.1` respectively `[::1]` are limited to the Container's own Network Namespace / Podman Pod and as such those will NOT resolve towards the Host Services/Ports.

**Comment by luckylinux** [comment]
Preface: this is while using `Podman` (Version 5.7.0, although any Version >= 5.0 should behave similarly).

In my Case it was actually a completely different Problem, but with some Requests going through correctly (e.g. `curl -L -vvv http://0.0.0.0:11434/api/version` yielding `{"version":"0.13.0"}`, `curl -L -vvv http://0.0.0.0:11434` yielding `Ollama is running`), some failing (e.g. `curl -L -vvv http://0.0.0.0:11434/models`).

Looking in `http://localhost:3000/admin/settings/models` I could see under **Ollama** `'http://0.0.0.0:11434'`.
**The single Quotes were the Issue** (it's NOT a Typo, that's how it was defined in the `.env` File).

Faulty `.env` File with `OLLAMA_BASE_URL='http://0.0.0.0:11434'`:
<img width="493" height="218" alt="image" src="https://github.com/user-attachments/assets/f1eb5a72-3fd9-47a6-a41e-7e627f93f803" />


Correct `.env` File with `OLLAMA_BASE_URL=http://0.0.0.0:11434`:
<img width="490" height="562" alt="image" src="https://github.com/user-attachments/assets/fbce35d7-413e-469b-9b85-162e86390da6" />


`ollama.pod` Quadlet:
```
[Pod]
PodName=ollama

# Ports
PublishPort=3000:8080/tcp
# PublishPort=11434:11434/tcp

# Network Mode
# Network=pasta
Network=ollama

# Map to Host's Localhost (only needed if want to access the Host **FROM** the Container)
# AddHost=host.docker.internal:host-gateway
# AddHost=host.containers.internal:host-gateway

[Install]
WantedBy=default.target
```

`ollama-server.container` Quadlet:
```
[Unit]
Description=Ollama Server Container
#Requires=ollama-caddy.service
#After=ollama-caddy.service

[Service]
Restart=always

[Container]
ContainerName=ollama-server

Pod=ollama.pod
StartWithPod=true

EnvironmentFile=.env
#Environment=OPEN_WEBUI_PORT=3000

Image=ollama/ollama

Pull=missing

#Network=container:ollama-caddy

SecurityLabelDisable=true
NoNewPrivileges=true
Pull=missing

#AddCapability=NET_RAW
#AddCapability=NET_BIND_SERVICE
#AddCapability=NET_ADMIN

AddDevice=nvidia.com/gpu=all

Volume=/home/podman/containers/data/ollama/server:/root/.ollama:rw,z
```

`ollama-open-webui.container` Quadlet:
```
[Unit]
Description=Ollama Open WebUI Container
Requires=ollama-server.service
After=ollama-server.service

[Service]
Restart=always

[Container]
ContainerName=ollama-open-webui

Pod=ollama.pod
StartWithPod=true

EnvironmentFile=.env
Environment=OPEN_WEBUI_PORT=3000

Image=ghcr.io/open-webui/open-webui:main

Pull=missing

#Network=container:ollama-server

NoNewPrivileges=true
Pull=missing

Volume=/home/podman/containers/data/ollama/open-webui:/app/backend/data:rw,z

```

**Comment by czar** [comment]
After updating Ollama on Ubuntu, Open WebUI lost its connection to `http://127.0.0.1:114434`. I resolved this by configuring Ollama to listen on all interfaces (`0.0.0.0`).

**Steps to fix:**
1. Run `sudo systemctl edit ollama` (or modify `/etc/systemd/system/ollama.service.d/override.conf`).
2. Add the following lines:
```ini
[Service]
Environment="OLLAMA_HOST=0.0.0.0"
```
3. Reload the daemon and restart the service:
```bash
sudo systemctl daemon-reload
sudo systemctl restart ollama
```
This resolves the network error and restores model access in Open WebUI.
